cpp/thirdparty/versions.txt

```diff
@@ -23,8 +23,8 @@
 # `DEPENDENCIES` array (see the comment on top of the declaration for the
 # format).
 
-ARROW_ABSL_BUILD_VERSION=20211102.0
-ARROW_ABSL_BUILD_SHA256_CHECKSUM=dcf71b9cba8dc0ca9940c4b316a0c796be8fab42b070bb6b7cab62b48f0e66c4
+ARROW_ABSL_BUILD_VERSION=20230125.2
```

Comment:
I'm not sure we can just update this without also possibly updating gRPC/Google Cloud SDK

Reply:
Hmm, seems Abseil promises compatibility now. So it looks like we're safe here, at least it compiles.